.github/workflows/build.yml

```diff
@@ -34,11 +34,7 @@ on:
 jobs:
   validate:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        java: [ 21, 17, 11, 8 ]
-    name: Compile and Check Java ${{ matrix.java }}
+    name: Compile and Check Java
```

Comment:
Since we are using the "release" property for our JavaCompile tasks, we actually don't need to run the compile check on all versions -- just the newest.

```
    // --release is the recommended way to select the target release, but it's only supported in Java 9 so we also
    // set --source and --target via `sourceCompatibility` and `targetCompatibility` a couple of lines below
    if (JavaVersion.current().isJava9Compatible())
      options.release = minJavaVersion
```